worksheet.rb

```diff
@@ -22,13 +44,104 @@
 # You should be copying and pasting the literal data
 # into this data structure, such as "DR0004"
 # and "3rd Feb 2016" and "RD0022"
+#
+# ------------- RESPONSE ----------------
+drivers ={"DR0001" => [{"DATE" => "3rd Feb 2016", "COST" => 10,"RIDER_ID" => "RD0003", "RATING" => 3},{"DATE" => "3rd Feb 2016", "COST" => 30, "RIDER_ID" => "RD0015", "RATING" => 4},{"DATE" => "5th Feb 2016", "COST" => 45, "RIDER_ID" => "RD0003", "RATING" => 2}],
+         "DR0002" => [{"DATE" => "3rd Feb 2016", "COST" => 25, "RIDER_ID" => "RD0073", "RATING" => 5},{"DATE" => "4th Feb 2016", "COST" => 15, "RIDER_ID" => "RD0013", "RATING" => 1},{"DATE" => "5th Feb 2016", "COST" => 35, "RIDER_ID" => "RD0066", "RATING" => 3}],
+         "DR0003" => [{"DATE" => "4th Feb 2016", "COST" => 5,"RIDER_ID" => "RD0066", "RATING" => 5},{"DATE" => "5th Feb 2016", "COST" => 50, "RIDER_ID" => "RD0003", "RATING" => 2}],
+         "DR0004" => [{"DATE" => "3rd Feb 2016", "COST" => 5,"RIDER_ID" => "RD0022", "RATING" => 5},{"DATE" => "4th Feb 2016", "COST" => 10, "RIDER_ID" => "RD0022", "RATING" => 4}, {"DATE" => "5th Feb 2016", "COST" => 20, "RIDER_ID" => "RD0073", "RATING" => 5}]}
 
 ########################################################
 # Step 4: Total Driver's Earnings and Number of Rides
 
 # Use an iteration blocks to print the following answers:
 # - the number of rides each driver has given
+
+# ------------- RESPONSE ----------------
+
+puts "ADA RIDE SHARE DATA INFORMATION"
+
+puts "\nThe number of rides each driver has given:"
+
+# iterate through first layer to find the size of each hash, indicating # of rides
+drivers.each do |driver, rides|
+  puts "DRIVER #{driver}: #{rides.length} RIDES"
+end
+
+
+
 # - the total amount of money each driver has made
+
+# ------------- RESPONSE ----------------
+
+puts "\nThe total amount of money each driver has made:"
+
+# creates a hash from the data structure where keys are the driver ID and values are calculate total money made from "COST" key per driver
+total_money = drivers.map{|driver, rides|{driver.to_s => rides.inject(0){|total, ride| total += ride["COST"]}}}.reduce({}, :merge)
```

Comment:
Interesting use of `:merge` to convert it from an array of hashes into one large hash.